apps/web/src/api/hooks/useGetSignedUrl.ts

```diff
@@ -0,0 +1,14 @@
+import { useMutation } from '@tanstack/react-query';
+import { IResponseError } from '@novu/shared';
+
+import { getSignedUrl, IGetSignedUrlParams, ISignedUrlResponse } from '../storage';
+
+export const useGetSignedUrl = () => {
+  const { mutateAsync: getSignedUrlMutation, ...mutationData } = useMutation<
+    ISignedUrlResponse,
+    IResponseError,
+    IGetSignedUrlParams
+  >(getSignedUrl);
+
+  return { getSignedUrl: getSignedUrlMutation, ...mutationData };
+};
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** Add the `onError` method with `errorMessage`‏

Reply:
@BiswaViraj I'll allow providing the options to this hook, because to me the `onError` should be added in the places where the hook will be used, and defining the implementation there will make this hook more flexible, for example, I can use it in another place with a different message, or even without a message.